controllers/datafeed_controller.py

```diff
@@ -453,14 +453,29 @@ def get(self, year):
 
 class EventListGet(webapp.RequestHandler):
     """
-    Fetch one year of events
-    FMSAPI should be trusted over FIRSTElasticSearch
+    Fetch all events for a given year via the FRC Events API.
     """
     def get(self, year):
         df = DatafeedFMSAPI('v2.0')
 
         fmsapi_events, event_list_districts = df.getEventList(year)
-        events = EventManipulator.createOrUpdate(fmsapi_events)
+        # All regular-season events can be inserted without any work involved.
+        # We need to de-duplicate offseason events from the FRC Events API with a different code than the TBA event code
+        fmsapi_events_offseason = [e for e in fmsapi_events if e.is_offseason]
+        events_to_put = set(fmsapi_events) - set(fmsapi_events_offseason)
+
+        matched_offseason_events, new_offseason_events = \
+            OffseasonEventHelper.categorize_offseasons(int(year), fmsapi_events_offseason)
+
+        # For all matched offseason events, make sure the FIRST code matches the TBA FIRST code
+        for tba_event, first_event in matched_offseason_events:
+            tba_event.first_code = first_event.event_short
+            events_to_put.append(tba_event) # Update TBA events - discard the FIRST event
```

Comment:
This seems half-done - do we want to update the underlying TBA event with the upstream FIRST event data? Does this get taken care of when we kickoff an `event_details` task for these events?

Reply:
I'm pretty sure this will be sufficient - the `event_details` task will use the `first_code` when fetching data from upstream.